### ISSUE TITLE
be sure that docs on swSpice() and [["spice"]] clearly distinguish EOS from gsw to unesco

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2188]
As discussed in #2187, if `a` is a CTD object then `swSpice(a)`, or equivalently `a[["spice"]]` yield different results from `a[["spiciness"]]`.  I need to look in the docs to be sure that this is clear enough not to surprise users.

**Comment by dankelley** [comment]
After some study of the docs and code, I realize that this is simply too confusing.  The problem is that the behaviour of `swSpice()` can depend on an option that is possibly defined **outside the user's script** and thus sharing scripts will be tricky.  But, after quite a bit of head scratching, I decided on a solution: *force* the `eos="unesco"` if the user does e.g. ``ctd[["spice"]]`.  That's actually the reasonable things, anyway, since that formula was designed for the old equation of state.  The new thing, called `spiciness` has a new name precisely to avoid confusion.  (Then I went ahead and confused things.)

I am doing this change and documenting it clearly.
